crates/oxc_ast/src/ast_impl/js.rs

```diff
@@ -1007,6 +1007,13 @@ impl<'a> Function<'a> {
     }
 }
 
+impl GetAddress for Function<'_> {
+    #[inline]
+    fn address(&self) -> Address {
+        Address::from_ptr(self)
+    }
+}
```

Comment:
Actually, there's a 3rd option. We could add a method `TraverseCtx::current_address` which would give you address of the `Function` in `exit_function`, while avoiding these problems.

That would require some changes to `Traverse`, which are possible but not entirely trivial. I can implement it if we expect to use `Address` in this way a lot, so it's a common problem which is worth solving.

Reply:
I got your meaning! Thanks for explaining a lot.

 > 1. Keep it, and accept that it's possible to make mistakes like this, and we have to be careful to avoid them. or

We are using our tests to guarantee we are used correctly 🫣

> Re-implement exit_function here as exit_statement instead, so then you have a &mut Box<'a, Function<'a>>, instead of plain &mut Function<'a>. Then we can remove impl GetAddress for Function - the existing impl GetAddress for Box<T> will cover it.

We need to handle these `function x() {}`, `export function x() {}` and `export default function x() {}` functions, If we handle in `exit_statements` which will makes many boilerplate code

> Actually, there's a 3rd option. We could add a method TraverseCtx::current_address which would give you address of the Function in exit_function, while avoiding these problems.


I like this approach

Reply:
> If we handle in `exit_statements` which will makes many boilerplate code.

Ah good point. OK, let's go with it as is.

I've opened an issue for adding `TraverseCtx::current_address`: https://github.com/oxc-project/backlog/issues/140